bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Rule.java

```diff
@@ -30,23 +32,123 @@
  * @author Ana Dimova - Initial Contribution
  * @author Vasil Ilchev - Initial Contribution
  */
-public interface Rule {
+public class Rule {
+
+    protected List<Trigger> triggers;
+    protected List<Condition> conditions;
+    protected List<Action> actions;
+    protected String scopeId;
+    protected Map<String, ?> configurations;
+    protected Set<ConfigDescriptionParameter> configDescriptions;
+    protected String ruleTemplateUID;
+    protected String uid;
+    private String name;
+    private Set<String> tags;
+    private String description;
+
+    public Rule() {
+    }
+
+    /**
+     * This constructor is used when creating the rule from template and there is not provided UID for the rule.
+     *
+     * @param ruleTemplateUID is the unique identifier of the template, used for creation of the rule.
+     * @param are values of the configuration parameters that are needed for configuring the rule, represented as pairs
+     *            key-value, where the key is the name of the configuration parameter and the value is its value.
+     */
+    public Rule(String ruleTemplateUID, Map<String, ?> configurations) {
+        this.ruleTemplateUID = ruleTemplateUID;
+        this.configurations = configurations;
+    }
+
+    /**
+     * This constructor is used when creating the rule from template and there is provided UID for the rule.
+     *
+     * @param uid is the unique identifier of the rule provided by its creator.
+     * @param ruleTemplateUID is the unique identifier of the template, used for creation of the rule.
+     * @param configurations are values of the configuration parameters that are needed for configuring the rule,
+     *            represented as pairs key-value, where the key is the name of the configuration parameter and the value
+     *            is its value.
+     */
+    public Rule(String uid, String ruleTemplateUID, Map<String, ?> configurations) {
+        this.uid = uid;
+        this.ruleTemplateUID = ruleTemplateUID;
+        this.configurations = configurations;
+    }
+
+    /**
+     * This constructor is used when creating the rule and there is not provided UID for the rule.
+     *
+     * @param triggers
+     * @param conditions
+     * @param actions
+     * @param configDescriptions
+     * @param configurations
+     */
+    public Rule(List<Trigger> triggers, //
+            List<Condition> conditions, //
+            List<Action> actions, //
+            Set<ConfigDescriptionParameter> configDescriptions, //
+            Map<String, ?> configurations) {
+        this.triggers = triggers;
```

Comment:
but why can null be set? I think null as value should not be allowed for module list. I would expect that the Rule always returns a list, neither empty or containing entries, but never null. So you can throw IllegalArgumentException, or in case null is given, just create an empty list by your self

> Also I'm not sure if it is good to create redundant objects

What do you mean with redundant object?